daily_milestone.go

```diff
@@ -166,21 +166,19 @@ func createMilestoneData(advance int, milestoneTime string) []simpleMilestone {
 			milestone.DueDate = date
 			list = append(list, milestone)
 		} else if milestoneTime == "weekly" {
-
 			today = today.AddDate(0, 0, 7)
 			year, week := today.ISOWeek()
 			milestoneYear := strconv.Itoa(year)
 			milestoneWeek := strconv.Itoa(week)
-			milestones := milestoneYear + "-w" + milestoneWeek
 			milestone := simpleMilestone{}
-			milestone.Title = milestones
-			milestone.DueDate = milestones
+			milestone.Title = milestoneYear + "-w" + milestoneWeek
+			milestone.DueDate = today.Format("2006-01-02")
 			list = append(list, milestone)
 		} else if milestoneTime == "monthly" {
 			date := today.AddDate(0, i, 0).Format("2006-01")
 			milestone := simpleMilestone{}
 			milestone.Title = date
-			milestone.DueDate = date
+			milestone.DueDate = today.Format("2006-01-02")
```

Comment:
For dueDate that is definitely the right format, but similar to the weekly case it has a wrong due date. It depends on the day you run dailymile. It needs to be set to the last day of the actual month.